src/base/cub_ddl_log.c

```diff
@@ -0,0 +1,1206 @@
+
+#include "porting.h"
+#include "cas_common.h"
+#include "cub_ddl_log.h"
+#include "parse_tree.h"
+#include "system_parameter.h"
+#include "environment_variable.h"
+#include "broker_config.h"
+
+#define DDL_LOG_BUFFER_SIZE         (8192)
+#define DDL_LOG_MSG 	            (256)
+#define DDL_LOG_PATH    	    "log/ddl_audit"
+#define DDL_LOG_LOADDB_FILE_PATH    "log/ddl_audit/loaddb"
+#define FILE_BUFFER_SIZE            (1024)
+#define TIME_STRING_SIZE            (16)
+#define DDL_TIME_LEN                (32)
+
+
+typedef struct t_ddl_audit_handle T_DDL_AUDIT_HANDLE;
+struct t_ddl_audit_handle
+{
+  char db_name[DB_MAX_IDENTIFIER_LENGTH];
+  char user_name[DB_MAX_USER_LENGTH];
+  T_APP_NAME app_name;
+  char br_name[BROKER_NAME_LEN];
+  int br_index;
+  char ip_addr[16];
+  int pid;
+  char *sql_text;
+  char stmt_type;
+  char execute_start_time[DDL_TIME_LEN];
+  char elapsed_time[DDL_TIME_LEN];
+  int file_line_number;
+  int err_code;
+  char *err_msg;
+  //char result;
+  char msg[DDL_LOG_MSG];
+  char file_name[PATH_MAX];
+  char schema_file[PATH_MAX];
+  char execute_type;
+  char loaddb_file_type;
+  char log_filepath[PATH_MAX];
+  int commit_count;
+  struct timeval exec_begin_time;
+  char auto_commit_mode;
+};
+static T_DDL_AUDIT_HANDLE *ddl_audit_handle = NULL;
+
+static int cub_make_ddl_log_filename (char *filename_buf, size_t buf_size, T_APP_NAME app_name);
+static int cub_make_schema_file_name (const char *file_full_path, char *dest_path, size_t buf_size);
+static void cub_ddl_log_backup (const char *path);
+#if defined(WINDOWS)
+static void unix_style_path (char *path);
+#endif /* WINDOWS */
+static int cub_create_dir_log (const char *new_dir);
+static int cub_create_log_mgs (char *msg);
+static int cub_get_current_time (char *buf, size_t size);
+static int cub_file_copy (char *src_file, char *dest_file);
+static void cub_remove_char (char *string, char ch);
+static FILE *cub_ddl_log_open (T_APP_NAME app_name);
+static int cub_get_time_string (char *buf, struct timeval *time_val);
+static FILE *cub_fopen_and_lock (const char *path, const char *mode);
+static void cub_ddl_log_elapsed_time (long sec, long msec);
+static void cub_timeval_diff (struct timeval *start, struct timeval *end, int *res_sec, int *res_msec);
+static const char *cub_get_app_name (T_APP_NAME app_name);
+
+static char is_executed_ddl = FALSE;
+
+void
+cub_ddl_log_init ()
+{
+  if (ddl_audit_handle != NULL)
+    {
+      FREE_MEM (ddl_audit_handle->sql_text);
+      FREE_MEM (ddl_audit_handle->err_msg);
+      FREE_MEM (ddl_audit_handle);
+    }
+
+  ddl_audit_handle = (T_DDL_AUDIT_HANDLE *) MALLOC (sizeof (T_DDL_AUDIT_HANDLE));
+  if (ddl_audit_handle == NULL)
+    {
+      return;
+    }
+
+  memset (ddl_audit_handle, 0x00, sizeof (T_DDL_AUDIT_HANDLE));
+
+  ddl_audit_handle->stmt_type = -1;
+  ddl_audit_handle->execute_type = DDL_LOG_RUN_EXECUTE_FUNC;
+  ddl_audit_handle->loaddb_file_type = LOADDB_FILE_TYPE_NONE;
+}
+
+void
+cub_ddl_log_free (char all_free)
```

Comment:
Types mismatched:
line, 660, 828, cas_function.c:2026, you called cub_ddl_log_free() in this style:
```
cub_ddl_log_free (FALSE);
cub_ddl_log_free (TRUE);
```

I recommend to change cub_ddl_log.c cub_ddl_log.h cas_function.c of this style
```
extern void cub_ddl_log_free (bool all_free);
void
cub_ddl_log_free (bool all_free)
cub_ddl_log_free (true);
cub_ddl_log_free (false);
```

Reply:
Thank you, i will change code as your recommend.